### ISSUE TITLE
Add Logistic Regression

### ISSUE THREAD

**Issue by param087** [param087/swiftML#6]
#### `LogisticRegression(iterationCount: 1000, learningRate: 0.1, fitIntercept: true)`
  - **iterationCount**: The maximum number of passes over the training data.
  - **learningRate**: The amount that the weights are updated during training.
  - **fitIntercept**: Whether to calculate the intercept for this model. If set to `false`, no intercept will be used in calculations.

#### API Methods
 **`fit(data: Tensor, labels: Tensor)`**: Fit a logistic regression model.
 **`prediction(for: Tensor)`**: Predict class for sample input data.
 **`score(data: Tensor, labels: Tensor)`**: Returns the mean accuracy.

###  **References** 
 - https://en.wikipedia.org/wiki/Logistic_regression
 - https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html
 - https://machinelearningmastery.com/logistic-regression-for-machine-learning/
 - http://kenzotakahashi.github.io/logistic-regression-from-scratch-in-python.html/
